Sources/Segment/Plugins/Platforms/Vendors/AppleUtils.swift

```diff
@@ -65,17 +66,15 @@ internal class iOSVendorSystem: VendorSystem {
     
     override var userAgent: String? {
         #if !os(tvOS)
-        var userAgent: String?
-        
-        if Thread.isMainThread {
-            userAgent = WKWebView().value(forKey: "userAgent") as? String
-        } else {
-            DispatchQueue.main.sync {
-              userAgent = WKWebView().value(forKey: "userAgent") as? String
+        // BKS: It was discovered that on some platforms there can be a delay in retrieval.
+        // It has to be fetched on the main thread, so we've spun it off
+        // async and cache it when it comes back.
+        if Self.asyncUserAgent == nil {
+            DispatchQueue.main.async {
```

Comment:
Hey any chance you should re-check the condition `asyncUserAgent == nil` once you're in there?  This reminds me of the patterns that would require double-check locking, no?

Reply:
I don't mind if it gets set twice.  It's locked via the @Atomic marker on the var so no race condition on double-set.